src/genai/_utils/general.py

```diff
@@ -184,11 +189,15 @@ def single_execution(fn: T) -> T:
     return functools.cache(fn)  # type: ignore
 
 
-def prompts_to_strings(prompts: Union[list[str], str, None]) -> list[str]:
+def prompts_to_strings(prompts: Union[Sequence[str], str, None]) -> Sequence[str]:
     if prompts is None:
         return []
 
-    if not isinstance(prompts, list):
+    if not is_sequence(prompts):
         return [prompts]
 
     return prompts
```

Comment:
We can remove this function completely.

Reply:
Not sure, wIth using just `cast_sequence` instead, it would lead to `prompts: Sequence[str | None]`